src/EditorFeatures/Core/Implementation/SmartIndent/SmartIndent.cs

```diff
@@ -40,21 +44,39 @@ public void Dispose()
             using (Logger.LogBlock(FunctionId.SmartIndentation_Start, cancellationToken))
             {
                 var document = lineToBeIndented.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
-                var syncService = document?.GetLanguageService<ISynchronousIndentationService>();
+                if (document == null)
+                {
+                    return null;
+                }
 
-                if (syncService != null)
+                // First, try to go through the normal feature-layer indentation service.
+                var newService = document.GetLanguageService<NewIndentationService>();
+                if (newService != null)
                 {
-                    var result = syncService.GetDesiredIndentation(document, lineToBeIndented.LineNumber, cancellationToken);
+                    var result = newService.GetDesiredIndentation(document, lineToBeIndented.LineNumber, cancellationToken);
                     return result?.GetIndentation(_textView, lineToBeIndented);
                 }
 
-                var asyncService = document?.GetLanguageService<IIndentationService>();
-                if (asyncService != null)
+                // If we don't have a feature-layer service, try to fall back to the legacy
+                // editor-feature-layer interfaces.
+
```

Comment:
is this for things like typescript or F#?

Reply:
Yup!